### ISSUE TITLE
DB::Exception: Cannot read all marks from file xxx/xxx/data, eof: 0, buffer size: 6866, file size: 176688: While executing MergeTreeThread SQLSTATE: 22000.

### ISSUE THREAD

**Issue by xiaomo728** [ByConity/ByConity#80]
I was just using ClickHouse official's **[Star Schema Benchmark](https://clickhouse.com/docs/en/getting-started/example-datasets/star-schema/)** to test the ByConity.
When I tried to create a big flat table lineorder_flat (about 40 columns) by INNER JOIN 4 tables (the biggest table has about 600 million rows), I got this exception:

`Received exception from server (version 21.8.7):
Code: 33. DB::Exception: Received from 127.0.0.1:56871. DB::Exception: Received from 127.0.0.1:53749. DB::Exception: Received from 127.0.0.1:41655. DB::Exception: Cannot read all marks from file c1133162-7809-4269-8ad5-41243831e384/1996_439329587933478912_439330151713472512_2_439330217603104768_0/data, eof: 0, buffer size: 6866, file size: 176688: While executing MergeTreeThread SQLSTATE: 22000.`

It seems to be a memory issue. I have not seen this error before when using ClickHouse. I assume this is a new error added to ByConity? Is this a problem with the HDFS related configuations or do I need to configure several ByConity settings correctly?

Thank you very much.

**Comment by canhld94** [comment]
Hi @xiaomo728 thank you for trying Byconity. We can reproduce this issue and will fix it asap, will let you know soon.

In addition, can I have some questions:

1. How're you deploy Byconity? We're continuously merge new performance optimizations, so it's recommended to use latest version for performance testing. If you're using docker/k8s deployment, let me update the newest image version.

2. Specially for SSB testing, I have some recommendations:
a. Before creating flat table, let's build the statistic of the table. This allow our CBO to work more efficient, and the join performance will be much better. To build statistic for all table (for example, your tables are in database `ssb100`), you can run: (1) `use ssb100`, then (2) `create stats if not exists all`. This will create statistics for all tables in current database. Note that the second query will take a while, but you only need to run it once :)
b. While creating flat table, let's `set enable_optimizer=1` to use CBO
c. In addition to flat queries, you can also test the Star Schema queries (JOIN queries) to compare with original Clickhouse (hint: it's an order of magnitude faster, and use much less memory)

**Comment by zhbdesign** [comment]
create stats if not exists all，is  OK？

**Comment by hustnn** [comment]
@xiaomo728 Have you solved the TSO issue?

**Comment by xiaomo728** [comment]
@hustnn Yes, I think I have solved that. I just re-installed the fdb. Many thx.

**Comment by hustnn** [comment]
@xiaomo728 Besides pull newer version, you can comment the second condition in line 132 of file `MergeTreeMarksLoader.cpp` to bypass this error and continue your testing, the root cause is clear now and it is related to this check.
<img width="1113" alt="image" src="https://user-images.githubusercontent.com/73186/217831272-7b494ace-df53-4bce-870e-2e10f892b6c7.png">

**Comment by hustnn** [comment]
> @hustnn Yes, I think I have solved that. I just re-installed the fdb. Many thx.

Please help to close related ticket once solved. Thanks.

**Comment by xiaomo728** [comment]
> > @hustnn Yes, I think I have solved that. I just re-installed the fdb. Many thx.
> 
> Please help to close related ticket once solved. Thanks.

@hustnn done.

**Comment by xiaomo728** [comment]
@canhld94 Hi, since I used the `create stats if not exists all`, I got this exception:

```
CREATE STATS IF NOT EXISTS ALL

Query id: 4b82a5d7-4a03-4ff4-9228-9ce1aae3cdcd

┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
│ cnch_test  │            2 │ 1                  │  0.099356323 │
└────────────┴──────────────┴────────────────────┴──────────────┘
┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
│ customer   │            8 │ 3000000            │  1.365814247 │
└────────────┴──────────────┴────────────────────┴──────────────┘
┌─table_name─┬─column_count─┬─row_count_or_error──────────────────────────────────────────────────────────────────────────────┬─elapsed_time─┐
│ lineorder  │           17 │ Code: 49, e.displayText() = DB::Exception: Not a Valid Block SQLSTATE: HY000 (version 21.8.7.1) │  1.350792488 │
└────────────┴──────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────┘
┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
│ part       │            9 │ 1250000            │  0.493515223 │
└────────────┴──────────────┴────────────────────┴──────────────┘
┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
│ supplier   │            7 │ 200000             │   0.46286751 │
└────────────┴──────────────┴────────────────────┴──────────────┘
↗ Progress: 612.09 million rows, 1.23 GB (148.65 million rows/s., 298.66 MB/s.) █████████████████████████████████████████████████████████████████▏                                                                49%
5 rows in set. Elapsed: 4.356 sec. Processed 612.09 million rows, 1.23 GB (140.52 million rows/s., 282.33 MB/s.)

Received exception from server (version 21.8.7):
Code: 117. DB::Exception: Received from 127.0.0.1:60447. DB::Exception: when collecting table lin.lineorder having the following error: Code: 49, e.displayText() = DB::Exception: Not a Valid Block SQLSTATE: HY000,
```

Tbh I don't know why could be like this. Now I am re-generating the test data and planing to try it again later.

**Comment by canhld94** [comment]
> @canhld94 Hi, since I used the `create stats if not exists all`, I got this exception:
> 
> ```
> CREATE STATS IF NOT EXISTS ALL
> 
> Query id: 4b82a5d7-4a03-4ff4-9228-9ce1aae3cdcd
> 
> ┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
> │ cnch_test  │            2 │ 1                  │  0.099356323 │
> └────────────┴──────────────┴────────────────────┴──────────────┘
> ┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
> │ customer   │            8 │ 3000000            │  1.365814247 │
> └────────────┴──────────────┴────────────────────┴──────────────┘
> ┌─table_name─┬─column_count─┬─row_count_or_error──────────────────────────────────────────────────────────────────────────────┬─elapsed_time─┐
> │ lineorder  │           17 │ Code: 49, e.displayText() = DB::Exception: Not a Valid Block SQLSTATE: HY000 (version 21.8.7.1) │  1.350792488 │
> └────────────┴──────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────┘
> ┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
> │ part       │            9 │ 1250000            │  0.493515223 │
> └────────────┴──────────────┴────────────────────┴──────────────┘
> ┌─table_name─┬─column_count─┬─row_count_or_error─┬─elapsed_time─┐
> │ supplier   │            7 │ 200000             │   0.46286751 │
> └────────────┴──────────────┴────────────────────┴──────────────┘
> ↗ Progress: 612.09 million rows, 1.23 GB (148.65 million rows/s., 298.66 MB/s.) █████████████████████████████████████████████████████████████████▏                                                                49%
> 5 rows in set. Elapsed: 4.356 sec. Processed 612.09 million rows, 1.23 GB (140.52 million rows/s., 282.33 MB/s.)
> 
> Received exception from server (version 21.8.7):
> Code: 117. DB::Exception: Received from 127.0.0.1:60447. DB::Exception: when collecting table lin.lineorder having the following error: Code: 49, e.displayText() = DB::Exception: Not a Valid Block SQLSTATE: HY000,
> ```
> 
> Tbh I don't know why could be like this. Now I am re-generating the test data and planning to try it again later.

Hi @xiaomo728 

Can you try to run `CREATE STATS IF NOT EXISTS ALL` again to see if it's just an intermittent error? Our people will also look at this issue.

Thank you.

**Comment by canhld94** [comment]
@xiaomo728 the exception during `CREATE STATS IF NOT EXISTS ALL` is likely caused by the exception of the internal SQL (used to collect statistics from the table). I think the internal SQL also got the same exception `Cannot read all marks from file` you described above.

We've already found the root cause of the issue and will merge the fix soon. Before that, to unblock your testing, you can (1) pull the newest codebase, and (2) make a minor change in MergeTreeMarksLoader.cpp as follows and recompile the binary.
Original:
https://github.com/ByConity/ByConity/blob/2f31a257dc5c7aa96864b5f699e75224df32e6ae/src/Storages/MergeTree/MergeTreeMarksLoader.cpp#L132

Let's change to:
```cpp
        if (buffer->eof())
```

Thanks.

**Comment by xiaomo728** [comment]
@canhld94 Hi, the logs of `Code: 49, e.displayText() = DB::Exception: Not a Valid Block SQLSTATE: HY000 (version 21.8.7.1)` exception caused by `CREATE STATS IF NOT EXISTS ALL` are as following:

(`xx` is just my file path)

```
2023.02.10 09:48:15.411582 [ 2524357 ] {1e8b9930-fb32-477f-9bd7-9103dd9f5c4b} <Error> CreateStats: Code: 49, e.displayText() = DB::Exception: Not a Valid Block SQLSTATE: HY000, Stack trace (when copying this message, always include the lines below):

0. Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1874f632 in /xx/xx/xxx/ByConity/bin/clickhouse
1. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0xae247e0 in /xx/xx/xxx/ByConity/bin/clickhouse
2. DB::Statistics::getOnlyRowFrom(DB::Statistics::SubqueryHelper&) @ 0x14e8ec20 in /xx/xx/xxx/ByConity/bin/clickhouse
3. DB::Statistics::StatisticsCollectorStepFull::collectFirstStep(std::__1::vector<DB::NameAndTypePair, std::__1::allocator<DB::NameAndTypePair> > const&) @ 0x14e8f7b6 in /xx/xx/xxx/ByConity/bin/clickhouse
4. DB::Statistics::StatisticsCollectorStepFull::collect(std::__1::vector<DB::NameAndTypePair, std::__1::allocator<DB::NameAndTypePair> > const&) @ 0x14e8f5fd in /xx/xx/xxx/ByConity/bin/clickhouse
5. DB::Statistics::StatisticsCollector::collect(std::__1::vector<DB::NameAndTypePair, std::__1::allocator<DB::NameAndTypePair> > const&) @ 0x14e8caf9 in /xx/xx/xxx/ByConity/bin/clickhouse
6. DB::collectStatsOnTarget(std::__1::shared_ptr<DB::Context const>, DB::Statistics::CollectorSettings const&, DB::CollectTarget const&) @ 0x13af0522 in /xx/xx/xxx/ByConity/bin/clickhouse
7. DB::(anonymous namespace)::CreateStatsBlockInputStream::readImpl() @ 0x13af3090 in /xx/xx/xxx/ByConity/bin/clickhouse
8. DB::IBlockInputStream::read() @ 0x1359c845 in /xx/xx/xxx/ByConity/bin/clickhouse
9. DB::AsynchronousBlockInputStream::calculate() @ 0x135984e4 in /xx/xx/xxx/ByConity/bin/clickhouse
10. void std::__1::__function::__policy_invoker<void ()>::__call_impl<std::__1::__function::__default_alloc_func<DB::AsynchronousBlockInputStream::next()::$_0, void ()> >(std::__1::__function::__policy_storage const*) @ 0x135987d0 in /xx/xx/xxx/ByConity/bin/clickhouse
11. ThreadPoolImpl<ThreadFromGlobalPool>::worker(std::__1::__list_iterator<ThreadFromGlobalPool, void*>) @ 0xae5f7cf in /xx/xx/xxx/ByConity/bin/clickhouse
12. ThreadFromGlobalPool::ThreadFromGlobalPool<void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda0'()>(void&&, void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda0'()&&...)::'lambda'()::operator()() @ 0xae6156c in /xx/xx/xxx/ByConity/bin/clickhouse
13. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0xae5c2c5 in /xx/xx/xxx/ByConity/bin/clickhouse
14. void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda0'()> >(void*) @ 0xae607da in /xx/xx/xxx/ByConity/bin/clickhouse
15. ? @ 0x8f4b in /usr/lib64/libpthread-2.28.so
16. __clone @ 0xf874f in /usr/lib64/libc-2.28.so
 (version 21.8.7.1)

```

I tried to execute this SQL many times, but all of them ended up with this exception and never finished.

Thank you guys so many valuable helps! I will keep giving feedback about ByConity to this community.

**Comment by hustnn** [comment]
@xiaomo728 Can you pull the latest version, this issue is fixed.

**Comment by xiaomo728** [comment]
> @xiaomo728 Can you pull the latest version, this issue is fixed.

@hustnn Sure. Many thx.

BTW - I am having issues with my FoundationDB again, the following is the status - is this caused by the lack of space of  operating space? Why is there a sudden lack of space? How could I fix it?

```
[$]# fdbcli
Using cluster file `fdb.cluster'.

The database is available, but has issues (type 'status' for more information).

Welcome to the fdbcli. For help, type `help'.
fdb> status;
>>> status

Using cluster file `fdb.cluster'.

Unable to start batch priority transaction after 5 seconds.

Configuration:
  Redundancy mode        - single
  Storage engine         - memory-2
  Coordinators           - 1
  Usable Regions         - 1

Cluster:
  FoundationDB processes - 1
  Zones                  - 1
  Machines               - 1
  Memory availability    - 8.0 GB per process on machine with least available
  Fault Tolerance        - 0 machines
  Server time            - 02/13/23 15:00:53

Data:
  Replication health     - (Re)initializing automatic data distribution
  Moving data            - unknown (initializing)
  Sum of key-value sizes - unknown
  Disk space used        - 138 MB

Operating space:
  Storage server         - 0.1 GB free on most full server
  Log server             - 0.0 GB free on most full server

Workload:
  Read rate              - 17 Hz
  Write rate             - 0 Hz
  Transactions started   - 5 Hz
  Transactions committed - 0 Hz
  Conflict rate          - 0 Hz
  Performance limited by server 909860e9e403a6a2: Log server running out of space (approaching 100MB limit).

Backup and DR:
  Running backups        - 0
  Running DRs            - 0

Client time: 02/13/23 15:00:48
```

**Comment by canhld94** [comment]
> > @xiaomo728 Can you pull the latest version, this issue is fixed.
> 
> @hustnn Sure. Many thx.
> 
> BTW - I am having issues with my FoundationDB again, the following is the status - is this caused by the lack of space of operating space? Why is there a sudden lack of space? How could I fix it?
> 
> ```
> [$]# fdbcli
> Using cluster file `fdb.cluster'.
> 
> The database is available, but has issues (type 'status' for more information).
> 
> Welcome to the fdbcli. For help, type `help'.
> fdb> status;
> >>> status
> 
> Using cluster file `fdb.cluster'.
> 
> Unable to start batch priority transaction after 5 seconds.
> 
> Configuration:
>   Redundancy mode        - single
>   Storage engine         - memory-2
>   Coordinators           - 1
>   Usable Regions         - 1
> 
> Cluster:
>   FoundationDB processes - 1
>   Zones                  - 1
>   Machines               - 1
>   Memory availability    - 8.0 GB per process on machine with least available
>   Fault Tolerance        - 0 machines
>   Server time            - 02/13/23 15:00:53
> 
> Data:
>   Replication health     - (Re)initializing automatic data distribution
>   Moving data            - unknown (initializing)
>   Sum of key-value sizes - unknown
>   Disk space used        - 138 MB
> 
> Operating space:
>   Storage server         - 0.1 GB free on most full server
>   Log server             - 0.0 GB free on most full server
> 
> Workload:
>   Read rate              - 17 Hz
>   Write rate             - 0 Hz
>   Transactions started   - 5 Hz
>   Transactions committed - 0 Hz
>   Conflict rate          - 0 Hz
>   Performance limited by server 909860e9e403a6a2: Log server running out of space (approaching 100MB limit).
> 
> Backup and DR:
>   Running backups        - 0
>   Running DRs            - 0
> 
> Client time: 02/13/23 15:00:48
> ```

_is this caused by the lack of space of operating space_ -- yes, probably the disk you used to store foundationdb data has running out of space. You can check disk space by command `df -h`.

We've meet this issue many times. For our case, the default location for foudationdb data is at `/var/` but `/` always almost full:
<img width="672" alt="image" src="https://user-images.githubusercontent.com/22796953/218409102-6c2d07a3-65e9-4256-bc5a-b366de63307a.png">

The solution is to migrate foundationdb data from `/` to another disk with more space (in our case, `/data00/`).

**Comment by xiaomo728** [comment]
@canhld94 Thanks. However, how should I change the space path?
I just find the `foundationdb.conf` file under the `/etc/foundationdb` and I think that this is the core config file of fdb. Thus I edited the `datadir` and `logdir` in this file to get a larger space and then restarted the fdb.
Unfortunately, it does not seem to work. Even my fdb will always be in the **UNAVAILABLE** state:
```
$ fdbcli
Using cluster file `/etc/foundationdb/fdb.cluster'.

The database is unavailable; type `status' for more information.

Welcome to the fdbcli. For help, type `help'.
fdb> status
;

Using cluster file `/etc/foundationdb/fdb.cluster'.

Could not communicate with a quorum of coordination servers:
  127.0.0.1:4500  (unreachable)

```

How should I change the space path correctly? Thank you very much.

**Comment by canhld94** [comment]
> @canhld94 Thanks. However, how should I change the space path?
> 
> I just find the `foundationdb.conf` file under the `/etc/foundationdb` and I think that this is the core config file of fdb. Thus I edited the `datadir` and `logdir` in this file to get a larger space and then restarted the fdb.
> 
> Unfortunately, it does not seem to work. Even my fdb will always be in the **UNAVAILABLE** state:
> 
> ```
> 
> $ fdbcli
> 
> Using cluster file `/etc/foundationdb/fdb.cluster'.
> 
> 
> 
> The database is unavailable; type `status' for more information.
> 
> 
> 
> Welcome to the fdbcli. For help, type `help'.
> 
> fdb> status
> 
> ;
> 
> 
> 
> Using cluster file `/etc/foundationdb/fdb.cluster'.
> 
> 
> 
> Could not communicate with a quorum of coordination servers:
> 
>   127.0.0.1:4500  (unreachable)
> 
> 
> 
> ```
> 
> 
> 
> How should I change the space path correctly? Thank you very much.

@dmthuc can you help to check the config of foundationdb, thanks 🙏.

**Comment by dmthuc** [comment]
> @canhld94 Thanks. However, how should I change the space path? I just find the `foundationdb.conf` file under the `/etc/foundationdb` and I think that this is the core config file of fdb. Thus I edited the `datadir` and `logdir` in this file to get a larger space and then restarted the fdb. Unfortunately, it does not seem to work. Even my fdb will always be in the **UNAVAILABLE** state:
> 
> ```
> $ fdbcli
> Using cluster file `/etc/foundationdb/fdb.cluster'.
> 
> The database is unavailable; type `status' for more information.
> 
> Welcome to the fdbcli. For help, type `help'.
> fdb> status
> ;
> 
> Using cluster file `/etc/foundationdb/fdb.cluster'.
> 
> Could not communicate with a quorum of coordination servers:
>   127.0.0.1:4500  (unreachable)
> ```
> 
> How should I change the space path correctly? Thank you very much.

Hi @xiaomo728 , look like your fdb is not up and listen at 127.0.0.1:4500. I think we could check the status of fdb by `systemctl status fdb.serivce` command

**Comment by xiaomo728** [comment]
> > @canhld94 Thanks. However, how should I change the space path? I just find the `foundationdb.conf` file under the `/etc/foundationdb` and I think that this is the core config file of fdb. Thus I edited the `datadir` and `logdir` in this file to get a larger space and then restarted the fdb. Unfortunately, it does not seem to work. Even my fdb will always be in the **UNAVAILABLE** state:
> > ```
> > $ fdbcli
> > Using cluster file `/etc/foundationdb/fdb.cluster'.
> > 
> > The database is unavailable; type `status' for more information.
> > 
> > Welcome to the fdbcli. For help, type `help'.
> > fdb> status
> > ;
> > 
> > Using cluster file `/etc/foundationdb/fdb.cluster'.
> > 
> > Could not communicate with a quorum of coordination servers:
> >   127.0.0.1:4500  (unreachable)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > How should I change the space path correctly? Thank you very much.
> 
> Hi @xiaomo728 , look like your fdb is not up and listen at 127.0.0.1:4500. I think we could check the status of fdb by `systemctl status fdb.serivce` command

@dmthuc It is not running.

```
[$]# systemctl status fdb.service
Unit fdb.service could not be found.
```

I tried to start the fdb.service by `service foundationdb start` but it does not work.
Sorry I am a noob of fdb.

**Comment by dmthuc** [comment]
Hi @xiaomo728 , no problem, could you tell me how do you deploy FDB? How do you restart FDB after you change the config file? I think the issue happen after you restart FDB

**Comment by dmthuc** [comment]
Hi @xiaomo728 , let us firstly create a channel for support user, then we can use it to diagnose your issue

**Comment by xiaomo728** [comment]
> Hi @xiaomo728 , no problem, could you tell me how do you deploy FDB? I think the issue happen after you restart FDB. Actually the `systemctl status fdb.service` is for the os that use systemd like Debian, I don't know about your OS

Hi, I just download the `rpm` package of fdb's server and client, then use `rpm -Uvh` command to install it. My OS is CentOS.

**Comment by xiaomo728** [comment]
> Hi @xiaomo728 , let us firstly create a channel for support user, then we can use it to diagnose your issue

@dmthuc Sure. How should I do here?

**Comment by dmthuc** [comment]
Hi @xiaomo728 , then the command for checking status will be `systemctl status foundationdb.service`

**Comment by xiaomo728** [comment]
> Hi @xiaomo728 , then the command for checking status will be `systemctl status foundationdb.service`

```
@dmthuc 
systemctl status foundationdb.service
● foundationdb.service - FoundationDB Key-Value Store
   Loaded: loaded (/usr/lib/systemd/system/foundationdb.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2023-02-14 14:07:15 CST; 2h 43min ago
  Process: 3256683 ExecStart=/usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize (code=exited, status=0/SUCCESS)
 Main PID: 3256684 (fdbmonitor)
    Tasks: 5
   Memory: 183.3M
   CGroup: /system.slice/foundationdb.service
           ├─3256684 /usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize
           ├─3256685 /usr/lib/foundationdb/backup_agent/backup_agent --cluster-file /etc/foundationdb/fdb.cluster --logdir /opt/huawei/foundationdb/log
           └─3397884 /usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize

Feb 14 16:50:57 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
Feb 14 16:50:58 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
Feb 14 16:50:59 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
Feb 14 16:51:00 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
```
This path `/xx/xxx/foundationdb/log/` is what I specified in the `foundationdb.conf` file, and it looks like it canot write logs to it.

**Comment by xiaomo728** [comment]
> @xiaomo728 Can you pull the latest version, this issue is fixed.

@hustnn Hi, I pull down the latest version code, and use the **SAME** config files to deploy ByConity, a new issue comes when initialize **dm**:
`<Error> Application: DB::Exception: 112:[E111]Fail to connect Socket{id=0 addr=127.0.0.1:8223} (0x0x7f37068d9180): Connection refused [R1][E112]Not connected to 127.0.0.1:8223 yet, server_id=0 [R2][E112]Not connected to 127.0.0.1:8223 yet, server_id=0 [R3][E112]Not connected to 127.0.0.1:8223 yet, server_id=0 SQLSTATE: HY000`

I can get into the client, and also can create or insert values into table. However, when I tried to read values in tables, it got error:
```
 :) select * from cn;

SELECT *
FROM cn

Query id: a2038dd3-7316-4ab4-b549-734b19938ad3


0 rows in set. Elapsed: 0.022 sec. 

Received exception from server (version 21.8.7):
Code: 279. DB::Exception: Received from 127.0.0.1:37381. DB::Exception: All connection tries failed. Log: 

Code: 210, e.displayText() = DB::NetException: Connection refused (127.0.0.1:9001) SQLSTATE: 08000 (version 21.8.7.1)
Code: 210, e.displayText() = DB::NetException: Connection refused (127.0.0.1:9001) SQLSTATE: 08000 (version 21.8.7.1)
Code: 210, e.displayText() = DB::NetException: Connection refused (127.0.0.1:9001) SQLSTATE: 08000 (version 21.8.7.1)

: While executing Remote SQLSTATE: 08000. 
```

What is going on? I did not change the config since last time I can use ByConity correctly.

**Comment by dmthuc** [comment]
> > Hi @xiaomo728 , then the command for checking status will be `systemctl status foundationdb.service`
> 
> ```
> @dmthuc 
> systemctl status foundationdb.service
> ● foundationdb.service - FoundationDB Key-Value Store
>    Loaded: loaded (/usr/lib/systemd/system/foundationdb.service; enabled; vendor preset: disabled)
>    Active: active (running) since Tue 2023-02-14 14:07:15 CST; 2h 43min ago
>   Process: 3256683 ExecStart=/usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize (code=exited, status=0/SUCCESS)
>  Main PID: 3256684 (fdbmonitor)
>     Tasks: 5
>    Memory: 183.3M
>    CGroup: /system.slice/foundationdb.service
>            ├─3256684 /usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize
>            ├─3256685 /usr/lib/foundationdb/backup_agent/backup_agent --cluster-file /etc/foundationdb/fdb.cluster --logdir /opt/huawei/foundationdb/log
>            └─3397884 /usr/lib/foundationdb/fdbmonitor --conffile /etc/foundationdb/foundationdb.conf --lockfile /var/run/fdbmonitor.pid --daemonize
> 
> Feb 14 16:50:57 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
> Feb 14 16:50:58 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
> Feb 14 16:50:59 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
> Feb 14 16:51:00 host-10-31-40-59 fdbmonitor[3256684]: LogGroup="default" Process="backup_agent.1": ERROR: could not create trace log file `/xx/xxx/foundationdb/log/trace.127.0.0.1.45421.1676354835.JUYyg6.0.1.>
> ```
> 
> This path `/xx/xxx/foundationdb/log/` is what I specified in the `foundationdb.conf` file, and it looks like it canot write logs to it.

@xiaomo728 then i think you may need to check if that path is exist and writable by root user

**Comment by xiaomo728** [comment]
> @xiaomo728 then i think you may need to check if that path is exist and writable by root user

This path is exist and writable by root user. However, the file (like `trace.127.0.0.1.xxxxxx`) does not exist under this path.

**Comment by dmthuc** [comment]
> > @xiaomo728 then i think you may need to check if that path is exist and writable by root user
> 
> @dmthuc This path is exist and writable by root user. However, the file (like `trace.127.0.0.1.xxxxxx`) does not exist under this path.

Hi @xiaomo728 , I think you haven't able to restart the FDB service successfully, please try `systemctl restart foundationdb.service`

**Comment by xiaomo728** [comment]
> systemctl restart foundationdb.service

Hi, @dmthuc does not work.

```
[$]# systemctl restart foundationdb.service
[$]# fdbcli
Using cluster file `/etc/foundationdb/fdb.cluster'.

The database is unavailable; type `status' for more information.

Welcome to the fdbcli. For help, type `help'.
fdb> 

```

**Comment by dmthuc** [comment]
> > systemctl restart foundationdb.service
> 
> Hi, @dmthuc does not work.
> 
> ```
> [$]# systemctl restart foundationdb.service
> [$]# fdbcli
> Using cluster file `/etc/foundationdb/fdb.cluster'.
> 
> The database is unavailable; type `status' for more information.
> 
> Welcome to the fdbcli. For help, type `help'.
> fdb> 
> ```

look like the fdb is up now, I wonder have you copy the old log to the new log directory?

**Comment by dmthuc** [comment]
Hi @xiaomo728 , if you haven't copy old log to the new log then you need to do it and retry `systemctl restart foundationdb.service`

**Comment by xiaomo728** [comment]
> Hi @xiaomo728 , if you haven't copy old log to the new log then you need to do it and retry `systemctl restart foundationdb.service`

Hi @dmthuc , I think I have copied the logs from old path `/var/log/foundationdb`:
```
[$ foundationdb]# ll
total 345164
-rw-r--r-- 1 foundationdb foundationdb  4519890 Feb 13 15:27 trace.127.0.0.1.28402.1676271634.AwIuIx.0.1.xml
-rw-r--r-- 1 foundationdb foundationdb 13111284 Feb 13 04:55 trace.127.0.0.1.4019.1675845905.jrdMrE.1.88.xml
-rw-r--r-- 1 foundationdb foundationdb 13107154 Feb 13 06:14 trace.127.0.0.1.4019.1675845905.jrdMrE.1.89.xml
-rw-r--r-- 1 foundationdb foundationdb 13108389 Feb 13 07:34 trace.127.0.0.1.4019.1675845905.jrdMrE.1.90.xml
-rw-r--r-- 1 foundationdb foundationdb 13112476 Feb 13 08:55 trace.127.0.0.1.4019.1675845905.jrdMrE.1.91.xml
-rw-r--r-- 1 foundationdb foundationdb 13109394 Feb 13 10:14 trace.127.0.0.1.4019.1675845905.jrdMrE.1.92.xml
-rw-r--r-- 1 foundationdb foundationdb 13114606 Feb 13 11:33 trace.127.0.0.1.4019.1675845905.jrdMrE.1.93.xml
-rw-r--r-- 1 foundationdb foundationdb 13111470 Feb 13 12:53 trace.127.0.0.1.4019.1675845905.jrdMrE.1.94.xml
-rw-r--r-- 1 foundationdb foundationdb 13114522 Feb 13 14:13 trace.127.0.0.1.4019.1675845905.jrdMrE.1.95.xml
-rw-r--r-- 1 foundationdb foundationdb  7457029 Feb 13 14:58 trace.127.0.0.1.4019.1675845905.jrdMrE.1.96.xml
-rw-r--r-- 1 foundationdb foundationdb 13133129 Feb 13 13:04 trace.127.0.0.1.4500.1675845905.aDR1U9.2.410.xml
-rw-r--r-- 1 foundationdb foundationdb 13126634 Feb 13 13:22 trace.127.0.0.1.4500.1675845905.aDR1U9.2.411.xml
-rw-r--r-- 1 foundationdb foundationdb 13114549 Feb 13 13:39 trace.127.0.0.1.4500.1675845905.aDR1U9.2.412.xml
-rw-r--r-- 1 foundationdb foundationdb 13114308 Feb 13 13:56 trace.127.0.0.1.4500.1675845905.aDR1U9.2.413.xml
-rw-r--r-- 1 foundationdb foundationdb 13113424 Feb 13 14:13 trace.127.0.0.1.4500.1675845905.aDR1U9.2.414.xml
-rw-r--r-- 1 foundationdb foundationdb 13120304 Feb 13 14:31 trace.127.0.0.1.4500.1675845905.aDR1U9.2.415.xml
-rw-r--r-- 1 foundationdb foundationdb 13115197 Feb 13 14:48 trace.127.0.0.1.4500.1675845905.aDR1U9.2.416.xml
-rw-r--r-- 1 foundationdb foundationdb  8420734 Feb 13 14:58 trace.127.0.0.1.4500.1675845905.aDR1U9.2.417.xml
-rw-r--r-- 1 foundationdb foundationdb 13126837 Feb 13 15:17 trace.127.0.0.1.4500.1676271635.1QDEMA.0.1.xml
-rw-r--r-- 1 foundationdb foundationdb  7311266 Feb 13 15:27 trace.127.0.0.1.4500.1676271635.1QDEMA.0.2.xml
-rw-r--r-- 1 foundationdb foundationdb 13489889 Feb 14 03:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.14.xml
-rw-r--r-- 1 foundationdb foundationdb 13490704 Feb 14 04:08 trace.127.0.0.1.51900.1676273265.t44z4F.1.15.xml
-rw-r--r-- 1 foundationdb foundationdb 13498882 Feb 14 04:58 trace.127.0.0.1.51900.1676273265.t44z4F.1.16.xml
-rw-r--r-- 1 foundationdb foundationdb 13495337 Feb 14 05:50 trace.127.0.0.1.51900.1676273265.t44z4F.1.17.xml
-rw-r--r-- 1 foundationdb foundationdb 13488044 Feb 14 06:40 trace.127.0.0.1.51900.1676273265.t44z4F.1.18.xml
-rw-r--r-- 1 foundationdb foundationdb 13496410 Feb 14 07:31 trace.127.0.0.1.51900.1676273265.t44z4F.1.19.xml
-rw-r--r-- 1 foundationdb foundationdb 13487186 Feb 14 08:21 trace.127.0.0.1.51900.1676273265.t44z4F.1.20.xml
-rw-r--r-- 1 foundationdb foundationdb 13491776 Feb 14 09:12 trace.127.0.0.1.51900.1676273265.t44z4F.1.21.xml
-rw-r--r-- 1 foundationdb foundationdb  7846456 Feb 14 09:41 trace.127.0.0.1.51900.1676273265.t44z4F.1.22.xml
```

Here is the new log path:
```
[$ log]# ll
total 345108
-rw-r--r-- 1 root root  4519890 Feb 14 17:17 trace.127.0.0.1.28402.1676271634.AwIuIx.0.1.xml
-rw-r--r-- 1 root root 13111284 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.88.xml
-rw-r--r-- 1 root root 13107154 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.89.xml
-rw-r--r-- 1 root root 13108389 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.90.xml
-rw-r--r-- 1 root root 13112476 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.91.xml
-rw-r--r-- 1 root root 13109394 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.92.xml
-rw-r--r-- 1 root root 13114606 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.93.xml
-rw-r--r-- 1 root root 13111470 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.94.xml
-rw-r--r-- 1 root root 13114522 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.95.xml
-rw-r--r-- 1 root root  7457029 Feb 14 17:17 trace.127.0.0.1.4019.1675845905.jrdMrE.1.96.xml
-rw-r--r-- 1 root root 13133129 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.410.xml
-rw-r--r-- 1 root root 13126634 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.411.xml
-rw-r--r-- 1 root root 13114549 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.412.xml
-rw-r--r-- 1 root root 13114308 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.413.xml
-rw-r--r-- 1 root root 13113424 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.414.xml
-rw-r--r-- 1 root root 13120304 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.415.xml
-rw-r--r-- 1 root root 13115197 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.416.xml
-rw-r--r-- 1 root root  8420734 Feb 14 17:17 trace.127.0.0.1.4500.1675845905.aDR1U9.2.417.xml
-rw-r--r-- 1 root root 13126837 Feb 14 17:17 trace.127.0.0.1.4500.1676271635.1QDEMA.0.1.xml
-rw-r--r-- 1 root root  7311266 Feb 14 17:17 trace.127.0.0.1.4500.1676271635.1QDEMA.0.2.xml
-rw-r--r-- 1 root root 13489889 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.14.xml
-rw-r--r-- 1 root root 13490704 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.15.xml
-rw-r--r-- 1 root root 13498882 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.16.xml
-rw-r--r-- 1 root root 13495337 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.17.xml
-rw-r--r-- 1 root root 13488044 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.18.xml
-rw-r--r-- 1 root root 13496410 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.19.xml
-rw-r--r-- 1 root root 13487186 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.20.xml
-rw-r--r-- 1 root root 13491776 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.21.xml
-rw-r--r-- 1 root root  7846456 Feb 14 17:17 trace.127.0.0.1.51900.1676273265.t44z4F.1.22.xml

```

I just use `cp` command, though I do not know why the `total` shows difference (`345164` and `345108`).

**Comment by dmthuc** [comment]
> foundationdb

Hi @xiaomo728 , I just notice that you have to change permission and owner of the new folder to `foundationdb`. After you do that, remove all new log. Stop foundationdb service with `systemctl stop foundationdb.service`, the copy log again then start foundationdb service again with `systemctl start foundationdb.service`

**Comment by xiaomo728** [comment]
> > foundationdb
> 
> Hi @xiaomo728 , I just notice that you have to change permission and owner of the new folder to `foundationdb`. After you do that, remove all new log. Stop foundationdb service with `systemctl stop foundationdb.service`, the copy log again then start foundationdb service again with `systemctl start foundationdb.service`

Hi @dmthuc , thank you very much! It works!
